TCC.Lib/Command/ProcessStartHelper.cs

```diff
@@ -0,0 +1,179 @@
+            var standardOutputResults = new TaskCompletionSource<List<string>>();
+            var standardErrorResults = new TaskCompletionSource<List<string>>();
+
+            process.OutputDataReceived += (sender, e) => OnOutput(e, standardOutput, standardOutputResults);
+            process.ErrorDataReceived += (sender, e) => OnError(outputClassifier, e, standardError, standardInfos, standardErrorResults);
+
+            var tcs = new TaskCompletionSource<CommandResult>();
+
+            async void OnProcessExit(object sender, EventArgs args)
+            {
+                var stdout = await standardOutputResults.Task;
+                var stderr = await standardErrorResults.Task;
+
+                var result = new CommandResult
+                {
+                    Command = command,
+                    ExitCode = process.ExitCode,
+                    IsSuccess = standardError.Count == 0,
+                    Output = string.Join(Environment.NewLine, stdout),
+                    Errors = string.Join(Environment.NewLine, stderr),
+                    Infos = standardInfos,
+                };
+
+                tcs.TrySetResult(result);
+                process.Exited -= OnProcessExit;
+            }
+
+            process.Exited += OnProcessExit;
+
+            using (cancellationToken.Register(() =>
+            {
+                tcs.TrySetCanceled();
+                process.Exited -= OnProcessExit;
+                KillProcessAndChilds(process);
+            }))
+            {
+                try
+                {
+                    if (process.Start() == false)
+                    {
+                        tcs.TrySetException(new InvalidOperationException("Failed to start process"));
+                        process.Exited -= OnProcessExit;
+                    }
+                    else
+                    {
+                        if (process.WaitForExit((int)Math.Min(TimeSpan.FromMinutes(timeoutMinutes).TotalMilliseconds, int.MaxValue)) == false)
+                        {
+                            tcs.TrySetException(new InvalidOperationException("Process timedout"));
+                            process.Exited -= OnProcessExit;
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    tcs.TrySetException(e);
+                    process.Exited -= OnProcessExit;
+                }
+                process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
+                return tcs.Task;
+            }
+        }
+
+        private static void KillProcessAndChilds(Process process)
+        {
+            try
+            {
+                KillProcessAndChildren(process.Id);
+                if (!process.HasExited)
+                {
+                    process.Kill();
+                }
+            }
+            catch (InvalidOperationException)
+            {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS108)